kpatch-build/create-diff-object.c

```diff
@@ -2168,15 +2168,15 @@ static void kpatch_check_relocations(struct kpatch_elf *kelf)
 {
 	struct rela *rela;
 	struct section *sec;
-	Elf_Data *sdata;
+	int sec_size;
 
 	list_for_each_entry(sec, &kelf->sections, list) {
 		if (!is_rela_section(sec))
 			continue;
 		list_for_each_entry(rela, &sec->relas, list) {
 			if (rela->sym->sec) {
-				sdata = rela->sym->sec->data;
-				if (rela->addend > (int)sdata->d_size) {
+				sec_size = rela->sym->sec->uncompressed_size;
```

Comment:
No need for an extra variable here

Reply:
Also, please just call it `sec->size`, and replace all uses of sec->data_d_size and sec->sh.sh_size with it.

Reply:
> No need for an extra variable here

yet, remove the unneeded variable of sec_size

Reply:
> No need for an extra variable here

yet, the value sdata had deleted.